src/modules/NetworkLogger.ts

```diff
@@ -2,14 +2,30 @@ import type { RequestHandler } from '@apollo/client';
 
 import InstabugConstants from '../utils/InstabugConstants';
 import xhr, { NetworkData, ProgressCallback } from '../utils/XhrNetworkInterceptor';
-import { reportNetworkLog, isContentTypeNotAllowed } from '../utils/InstabugUtils';
+import { isContentTypeNotAllowed, reportNetworkLog } from '../utils/InstabugUtils';
+import { NativeModules } from 'react-native';
 
 export type { NetworkData };
 
 export type NetworkDataObfuscationHandler = (data: NetworkData) => Promise<NetworkData>;
 let _networkDataObfuscationHandler: NetworkDataObfuscationHandler | null | undefined;
 let _requestFilterExpression = 'false';
 
+const filterURL = (url: string): string => {
+  const [protocol, rest] = url.split('://');
+  return (rest || protocol).split('/')[0];
+};
+
+const getDevServerURL = (): string | null => {
+  if (NativeModules.SourceCode) {
```

Comment:
this probably wont work with the new architecture how about adding a fallback something like this :

`import Config from 'react-native-config';`
`const configURL = Config.DEV_SERVER_URL;`  
`if (configURL) {return configURL}`

what do you think ?

Reply:
@kholood-ea This is a very good point that we might need to validate.
Though I don't think we can use `react-native-config` in our SDK since it's not a dependency of our SDK (and we don't intend to add it), but we can probably check its implementation in `react-native-config` and use a similar approach if that's doable.
@ahmedAlaaInstabug What do you think?

Reply:
This code doesn’t work in the new architecture setup :( It only works if the user specifies the dev server in the **.env** file.

Reply:
@ahmedAlaaInstabug I believe we need to find an approach that works with the new architecture as well, what do you think?

Reply:
I updated the logic to check if the URL port matches the Metro port, and if so, it will be excluded. Additionally, I added a new API to set the Metro port if it’s changed from the default 8081.